src/core.jl

```diff
@@ -20,14 +25,18 @@ Optionally:
 
 See also [`AbstractExplicitContainerLayer`](@ref)
 """
-abstract type AbstractExplicitLayer end
+abstract type AbstractExplicitLayer{hasparams, hasstate} end
 
 """
     initialparameters(rng::AbstractRNG, l)
 
 Generate the initial parameters of the layer `l`.
 """
 initialparameters(::AbstractRNG, ::AbstractExplicitLayer) = NamedTuple()
+function initialparameters(::AbstractRNG,
+                           ::AbstractExplicitLayer{false, hasstate}) where {hasstate}
```

Comment:
Unused `hasstate`

Reply:
Fixed in [37f4851](https://github.com/avik-pal/Lux.jl/pull/84/commits/37f48519a731897b9de5771ccb714ff0eea2e013)